src/server.mjs

```diff
@@ -265,6 +265,31 @@ async function getChunkedData(req, res, next) {
   console.log(`region = ${req.body.region}`);
   console.log(`tracks = ${JSON.stringify(req.body.tracks)}`);
 
+  // This will have a conitg, start, end, or a contig, start, distance
+  let parsedRegion;
+  try {
+    parsedRegion = parseRegion(req.body.region);
+  } catch (e) {
+    // Whatever went wrong in the parsing, it makes the request bad.
+    throw new BadRequestError("Wrong query: " + e.message + " See ? button above.");
+  }
+
+  // There's a chance this request was sent before the proper tracks were fetched
+  // This can happen when the bed file is a url and track names need to be downloaded
+  // TODO: There's also a chance the user decided to use different tracks than what was provided in the bedFile,
+  // maybe we shouldn't always replace tracks if found?
+  if (req.body.bedFile && isValidURL(req.body.bedFile)) {
+    const chunk = await getChunkName(req.body.bedFile, parsedRegion);
+    const fetchedTracks = await getChunkTracks(req.body.bedFile, chunk);
+
+    // We're always replacing the given tracks if we were able to find tracks from the bed file
+    if (fetchedTracks) {
+      console.log("Using new fetched tracks", JSON.stringify(req.body.tracks));
+      req.body.tracks = fetchedTracks;
```

Comment:
I think we need to do something more complex here to avoid clobbering the color information, which is stored inside the tracks. If the client hasn't sent tracks, we should fill in the tracks with their default color settings. If the client has sent tracks *and* has set up their own color selections, we need to retain the client's color choices.

I think if we don't, the client will apply whatever tracks we set here, when the request comes back, and then display with the default colors instead of the user-selected colors, right?